OTVision/application/detect/detection_file_save_path_provider.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+
+from OTVision.application.get_current_config import GetCurrentConfig
+
+
+class DetectionFileSavePathProvider:
+    """Provides a mechanism to generate file save paths for detections.
+
+    This class is responsible for deriving the appropriate filenames for
+    detection files based on the source and current configuration
+    settings. It utilizes a configuration provider to dynamically fetch
+    the required configuration values.
+
+    Args:
+        get_current_config (GetCurrentConfig): Retrieve the current application
+            configuration.
+    """
+
+    def __init__(self, get_current_config: GetCurrentConfig) -> None:
+        self._get_current_config = get_current_config
+
+    def provide(self, source: str) -> Path:
+        config = self._get_current_config.get()
+        return derive_filename(
+            video_file=Path(source),
+            detect_suffix=config.filetypes.detect,
+            detect_start=config.detect.detect_start,
+            detect_end=config.detect.detect_end,
+        )
+
+
+def derive_filename(
```

Comment:
We can use OTVision in OTCloud as a library to reuse the `derive_filename` function. It is currently also defined in OTCloud.

Reply:
I created an Issue on open project addressing this. Please see https://openproject.platomo.de/projects/001-opentrafficcam-live/work_packages/7524/activity?query_id=740